### PR TITLE
Check strata input

### PR DESCRIPTION
**Pull request by hfrick**
Doing the thing!

- rlang type checker for #458
- thread the call through for  #541

### CODE REVIEW

File: R/misc.R [hunk 1]
```diff
@@ -98,16 +98,17 @@ add_class <- function(x, cls) {
   x
 }
 
-strata_check <- function(strata, data) {
+check_strata <- function(strata, data, call = caller_env()) {
```
Comment (EmilHvitfeldt) [inline R/misc.R:101]: love the rename!

File: R/misc.R [hunk 2]
```diff
@@ -98,16 +98,17 @@ add_class <- function(x, cls) {
...
-    if (!is.character(strata) | length(strata) != 1) {
-      cli_abort("{.arg strata} should be a single name or character value.")
-    }
     if (inherits(data[, strata], "Surv")) {
-      cli_abort("{.arg strata} cannot be a {.cls Surv} object. Use the time or event variable directly.")
-    }
-    if (!(strata %in% names(data))) {
-      cli_abort("{strata} is not in {.arg data}.")
+      cli_abort(c(
+        "{.arg strata} cannot be a {.cls Surv} object.",
```
Comment (EmilHvitfeldt) [inline R/misc.R:107]: Technically `strata` is a character or name. but it is used to refer to column in the data set. So I think `.field` would be better here as it refers to the value

```suggestion
        "{.field strata} cannot be a {.cls Surv} object.",
```

**Review by EmilHvitfeldt — APPROVED** [review]
Love it 🔥

File: R/misc.R (hunk 2 above)
Comment (hfrick) [reply R/misc.R:107]: Good point, thank you!

**Pull request merged**